src/tree.rs

```diff
@@ -217,53 +221,62 @@ where
         node_key: NodeKey,
         version: Version,
         existing_leaf_node: LeafNode,
-        kvs: &[(KeyHash, OwnedValue)],
+        kvs: &[(KeyHash, Option<OwnedValue>)],
         depth: usize,
         hash_cache: &Option<&HashMap<NibblePath, [u8; 32]>>,
         tree_cache: &mut TreeCache<R>,
-    ) -> Result<(NodeKey, Node)> {
+    ) -> Result<Option<(NodeKey, Node)>> {
```

Comment:
Same question as above, why are we returning an `Option` here?

Reply:
See: https://github.com/penumbra-zone/jmt/pull/52/files/51e2b5656b962f7feaaad73c2c00bf4cc25363e7#r984713964